integrations/real-time-collaboration.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Integration: Real-Time Collaboration.
+ *
+ * @package Automattic\VIP\Integrations
+ */
+
+namespace Automattic\VIP\Integrations;
+
+/**
+ * Loads Real-Time Collaboration VIP Integration.
+ *
+ * @private
+ */
+class RealTimeCollaborationIntegration extends Integration {
+
+	public function is_loaded(): bool {
+		// Check for the existence of the plugin version constant defined in the main plugin file.
+		return defined( 'VIP_REAL_TIME_COLLABORATION__LOADED' );
+	}
+
+	/**
+	 * Loads the plugin.
+	 *
+	 * This is called after the integration is activated and configured.
+	 *
+	 * @private
+	 */
+	public function load(): void {
+		/**
+		 * Return if the integration is already loaded.
+		 *
+		 * In activate() method we do make sure to not activate the integration if its already loaded
+		 * but still adding it here as a safety measure i.e. if load() is called directly.
+		 */
+		if ( $this->is_loaded() ) {
+			return;
+		}
+
+		/**
+		 * Load the custom build of Gutenberg from vip-integrations.
+		 *
+		 * Built from - https://github.com/Automattic/gutenberg/tree/add/experimental-collaborative-editing
+		 * This has the code required to support collaborative editing.
+		 */
+		$gutenberg_path = WPVIP_MU_PLUGIN_DIR . '/vip-integrations/gutenberg/gutenberg.php';
+		if ( file_exists( $gutenberg_path ) ) {
+			require_once $gutenberg_path;
+		}
+
+		/**
+		 * Get all the entries in the path of WPVIP_MU_PLUGIN_DIR/vip-integrations/vip-real-time-collaboration-<version>/
+		 * and check what versions are available.
+		 */
+		$versions = $this->get_versions();
```

Comment:
We should consider pulling this out into a utility, as the RDB plugin uses this as well.